app/src/marketplace/utils/smartContract.js

```diff
@@ -39,6 +43,13 @@ export const getPrefixedHexString = (hex: string): string => hex.replace(/^0x|^/
 
 export const getUnprefixedHexString = (hex: string): string => hex.replace(/^0x|^/, '')
 
+export const calculateContractAddress = async (account: Address): Promise<Address> => {
+    const web3 = getWeb3()
```

Comment:
Use public node? Or the modal should actually check that Metamask is open (wrap with `withWeb3`)